spec/index.bs

```diff
@@ -192,34 +192,81 @@ dictionary GPULimits {
 
-    void destroy();
-};
-</script>
+<dl dfn-type=attribute dfn-for="GPUBuffer">
+	: <dfn>\[[size]]</dfn>
+	::
+        The Length of the {{GPUBuffer}} allocation in bytes, which is an unsigned long long.
+
+	: <dfn>\[[usage]]</dfn>
+	::
+        The allowed usages for this {{GPUBuffer}}, represented as a bitset unsigned long.
+
+    : <dfn>\[[state]]</dfn>
+    ::
+        This {{GPUBuffer}}'s state which is one of `"unmapped"`, `"mapped"`, `"destroyed"`.
+</dl>
+
+## Creation ## {#buffer-creation}
 
-### Creation ### {#buffer-creation}
+A {{GPUBuffer}} can be created unmapped using the {{GPUDevice/createBuffer(descriptor)}} method.
+
+### {{GPUBufferDescriptor}} ### {#GPUBufferDescriptor}
+
+This specifies the options to use in creating a {{GPUBuffer}}.
+The {{GPUBufferDescriptor/size}} and {{GPUBufferDescriptor/usage}} members are required.
+A {{NotFoundError}} exception MUST be thrown if any of the required members are not specified.
 
 <script type=idl>
 dictionary GPUBufferDescriptor : GPUObjectDescriptorBase {
-    required u64 size;
+    required unsigned long long size;
     required GPUBufferUsageFlags usage;
 };
 </script>
 
+<dl dfn-type="abstract-op">
+    : <dfn>validating GPUBufferDescriptor</dfn>(device, descriptor)
+    ::
+        <div algorithm="validation GPUBufferDescriptor(device, descriptor)">
+            1. If device is lost return false.
+            1. If any of the bits of |descriptor|'s {{GPUBufferDescriptor/usage}} aren't present in this device's [[allowed buffer usages]] return false.
+            1. If both the {{GPUBufferUsage/MAP_READ}} and {{GPUBufferUsage/MAP_WRITE}} bits of |descriptor|'s {{GPUBufferDescriptor/usage}} attribute are set, return false.
+            1. Return true.
+        </div>
+</dl>
+
+### {{GPUDevice/createBuffer(descriptor)|GPUDevice.createBuffer(descriptor)}} ### {#GPUDevice-createBuffer}
+
+<dl dfn-type="method" dfn-for="GPUDevice">
+	: <dfn>createBuffer(descriptor)</dfn>
+	::
+		<div algorithm="GPUDevice.createBuffer(descriptor)">
+            1. If the result of [$validating GPUBufferDescriptor$](this, descriptor) is false:
+
+                1. Record a validation error on the current scope.
+                1. Let |b| be the result of creating an error buffer.
+                1. Return |b|.
+
+            1. Let |b| be a new {{GPUBuffer}} object.
+            1. Respectively assign the values of the attributes
+				{{GPUBufferDescriptor/size}} and {{GPUBufferDescriptor/usage}} of the {{GPUBufferDescriptor}}
+                passed in the constructor to the internal slots {{GPUBuffer/[[size]]}} and {{GPUBuffer/[[usage]]}} of |b|.
+            1. Set the {{GPUBuffer/[[state]]}} internal slot of |b| to `"unmapped"`.
+            1. Set each byte of |b|'s allocation to zero.
+            1. Return |b|.
+        </div>
+</dl>
+
 ## Buffer Usage ## {#buffer-usage}
 
 <script type=idl>
-typedef u32 GPUBufferUsageFlags;
+typedef unsigned long GPUBufferUsageFlags;
```

Comment:
A typedef for flags would be nice too, if we're going to have int typedefs.

(both of these can be done later)

Reply:
#423